azure/templates/pds-tests.yml

```diff
@@ -47,11 +47,12 @@ steps:
 
   - bash: |
       export APIGEE_API_TOKEN="$(secret.AccessToken)"
+      export APIGEE_ACCESS_TOKEN="$(secret.AccessToken)"
       export OAUTH_BASE_URI="https://$(ENVIRONMENT).api.service.nhs.uk"
-      export OAUTH_PROXY="oauth2"
+      export OAUTH_PROXY="oauth2-mock"
       export INTERNAL_DEV_ASID="$(INTERNAL_DEV_ASID)"
 
-      poetry run pytest -v tests/functional/test_application_restricted.py --junitxml=tests/functional-test-report.xml
+      poetry run pytest -v tests/functional/test_application_restricted.py --api-name=personal-demographics-service --proxy-name "$(FULLY_QUALIFIED_SERVICE_NAME)" --junitxml=tests/functional-test-report.xml --reruns 20 --reruns-delay 1
```

Comment:
Although I added `--reruns 20`, ideally we wouldn't have to run a test 20 times. Can we refactor the tests such that we do not need such a high rerun count? This may need chasing up with the support channel.